### PR TITLE
refactor(cli/tools): reduce cloning

### PR DESCRIPTION
**Pull request by GJZwiers**
This PR removes clones in some of the tools.

### CODE REVIEW

**Comment by bartlomieju** [comment]
@GJZwiers looks good, do you look for those manually or using some kind of clippy lint?

**Comment by GJZwiers** [comment]
I looked for them manually, though a lint for this would be awesome. There is https://rust-lang.github.io/rust-clippy/master/#redundant_clone (which I think is enabled by default) but it is pretty conservative and does not catch all of these.

**Comment by GJZwiers** [comment]
@bartlomieju I may have created some flaky watcher tests with these changes, please don't merge yet

File: cli/tools/test.rs [hunk 1]
```diff
@@ -1004,48 +1004,47 @@ async fn test_specifiers(
...
-          return Ok(());
-        }
+    specifiers_with_mode
+      .into_iter()
+      .map(move |(specifier, mode)| {
+        let ps = ps.clone();
+        let permissions = permissions.clone();
+        let mut sender = sender.clone();
+        let options = options.clone();
+        let fail_fast_tracker = FailFastTracker::new(options.fail_fast);
```
Comment (crowlKats) [inline cli/tools/test.rs:1018]: this seems like unnecessary indirection for no real benefit

**Review by crowlKats — COMMENTED** [review]
LGTM

File: cli/tools/test.rs (hunk 1 above)
Comment (GJZwiers) [reply cli/tools/test.rs:1018]: The tracker is not used anywhere else so I moved it inside the closure to avoid a clone

**Review by crowlKats — APPROVED** [review]
LGTM, thanks

**Pull request merged**